airsonic-main/src/main/java/org/airsonic/player/domain/MediaFile.java

```diff
@@ -471,12 +484,70 @@ public Integer apply(MediaFile from) {
         };
     }
 
+    public void setPathForSingleFileMedia(String mediaFilePath, int songStart, int songEnd) {
```

Comment:
This is a bit gross, wouldn't be possible to have this stored in a field somewhere instead?

Reply:
Maybe, will look later, no time now...

Reply:
While this might be 'gross' it does have the advantage of being the least invasive way to add this functionality. I can change it to something more 'pure' but that'd necessitate changes to a fair bit of infrastructure as well as a schema update.

Reply:
Hey @jvoisin and @Yetangitu, from what I'm understanding the main reason this is not merged is this discussion. Is that correct?

Reply:
This function looks different in the latest version of the patch, it is only used to make it possible to store tracks in the database while the actual start time has been moved to a new field and the end time is calculated by adding the duration to the start time.

I don't know why it has not been merged, I more or less dropped the issue and went on with my life and other projects. I have been running a patched version for a long time now so I know the thing works as intended so that can not be the reason for this patch to remain in limbo.

I just noticed that the patch is suffering from bit rot, not that strange given the fact that I haven't touched it. If the maintainers are not interested in merging it I'll just keep my own patched version up to date. @jvoisin, @muff1nman, @eharris  - what about it? I requested a review several months ago but nothing has happened. If there is no interest in this feature or in my implementation I'd say just close this PR, otherwise let me know so I can update the patch. In that case I'd like for it to be merged within a reasonable time frame so it doesn't accrue more bit rot - I'm a bit short on time so I probably won't be able to keep it up to date all the time. Your call, I'm fine with keeping a patched/forked version running for my own needs.

Reply:
Let's resolve this for now. As long as media_file.path is designated as unique (see https://github.com/airsonic/airsonic/pull/856#issuecomment-531203448) the _gross_ mangling of this field is needed to allow indexed tracks to be added to the database. Either change the database scheme so this field is no longer unique and tracks are identified by media_file.path + media_file.start_position or accept that this _gross_ scheme is in fact needed.